lib/Frontend/CompilerInvocation.cpp

```diff
@@ -640,7 +640,8 @@ bool FrontendArgsToOptionsConverter::computeModuleName() {
       (Opts.ModuleName != STDLIB_NAME || Opts.ParseStdlib)) {
     return false;
   }
-  if (!Opts.actionHasOutput() || Opts.isCompilingExactlyOneSwiftFile()) {
+  if (FrontendOptions::doesActionImplyMainModule(Opts.RequestedAction) ||
```

Comment:
I approve of not pretending this predicate is for anything else, but I do have an issue with the name after all. :-) "Main module" is already a term we use to mean "the module being compiled".

After looking at the conditions under which we actually use this name, it seems that it's only in cases where the module name doesn't matter very much, because we don't expect to emit any output with mangled names in it or because we're immediately going to run the code. Maybe invert it again and call it `needsProperModuleName`?

Reply:
Yes! Done.